### PR TITLE
feat: solution to query the mirror node for the account balance, account info, and contract info data

### LINKED ISSUE #1697: Add Account Balance/Info, Contract Info Mirror Node Queries
### Problem

Detailed: https://github.com/hashgraph/hedera-sdk-reference/issues/144

### Solution

Add a service to query balance data for the corresponding network:

Detailed:
https://www.notion.so/limechain/Mirror-Node-Queries-fe0d563db0d545fdafd7a0a5137fe5bd

HIP: https://hips.hedera.com/hip/hip-873

Mirror Node: https://github.com/hashgraph/hedera-mirror-node/issues/7386

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by thenswan**
**Description**:
This PR adds a solution to query the mirror node for the account balance, account info, and contract info data.

**Related issue(s)**:
Fixes #1697 

**Notes for reviewer**:
- What’s changed
    - Introduced `MirrorNodeRouter`, `MirrorNodeGateway` and `MirrorNodeService` to interact with the Mirror Node via REST API.
    - Added new queries to the mirror node related to HIP-367.
    - Refactored old queries to the mirror node related to populating account num, account EVM address, etc.
    - Updated `AccountBalanceQuery`, `AccountInfoQuery` and `ContractInfoQuery` to fetch some data from the mirror node (per HIP-367).
    - Added `wiremock` dependency for unit tests. `MockingTest` relies on updated queries, so there is a need to mock the REST API as well.
    - Added `Thread#sleep()` to integration tests and examples (to wait until the mirror node updates with the new data). This will increase the time for CI checks.
    - Updated `ClientTest` to make it work as intended (not rely on the local node).
- Other points
    - Added `LOCAL_NODE_PORT` static variable to the `MirrorNodeRouter` specifically for `MockingTest`. Since we now have interaction with the mirror node via the REST API (previously we only had interactions via gRPC), there is a need to add an SDK API to configure it to step over this temporary solution.
    - Address points from the sonarcloud?

**Checklist**
- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: sdk/src/main/java/com/hedera/hashgraph/sdk/AccountBalanceQuery.java [hunk 1]
```diff
@@ -127,7 +134,40 @@ void onMakeRequest(com.hedera.hashgraph.sdk.proto.Query.Builder queryBuilder, Qu
...
+
+        List<TokenBalance> tokenBalanceList = tokens.asList().stream().map(jsonElement -> {
+            var jsonObject = jsonElement.getAsJsonObject();
+            var tokenId = jsonObject.get("token_id").getAsString();
+            var tokenBalance = jsonObject.get("balance").getAsLong();
+
+            return TokenBalance.newBuilder()
+                .setTokenId(TokenId.fromString(tokenId).toProtobuf())
+                .setBalance(tokenBalance)
+                // no tokenDecimals :( -- additional query per each token is needed to figure it out
```
Comment (thenswan) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/AccountBalanceQuery.java:160]: @ochikov 👀

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeGateway.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    JsonObject getAccountTokens(String idOrAliasOrEvmAddress) throws IOException, InterruptedException {
+        var fullApiUrl = MirrorNodeRouter.buildApiUrl(mirrorNodeUrl, ACCOUNT_TOKENS_ROUTE, idOrAliasOrEvmAddress);
+
+        var responseBody = performQueryToMirrorNode(fullApiUrl);
+
+        return JsonParser.parseString(responseBody).getAsJsonObject();
+    }
+
+    private String performQueryToMirrorNode(String apiUrl) throws IOException, InterruptedException {
+        HttpClient httpClient = HttpClient.newHttpClient();
```
Comment (0xivanov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeGateway.java:84]: httpClient could be extracted as final variable, so it's not initialized every time

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+ * limitations under the License.
+ *
+ */
+package com.hedera.hashgraph.sdk;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+class MirrorNodeRouter {
```
Comment (0xivanov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java:26]: Can add private constructor, as this is essentially static/util class.

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    private static final String API_VERSION = "/api/v1";
+
+    // refactor me
+    static String LOCAL_NODE_PORT = "5551";
```
Comment (0xivanov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java:31]: Can this be made configurable via some config file or something like this?

Comment (thenswan) [reply sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java:31]: @0xivanov Unfortunately, no, I mentioned it under "Other points" in the PR description.

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java (hunk 3 above)
Comment (thenswan) [reply sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeRouter.java:26]: @0xivanov addressed it.

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeGateway.java (hunk 2 above)
Comment (thenswan) [reply sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeGateway.java]: @0xivanov addressed it.

**Review by 0xivanov — APPROVED** [review]
LGTM

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeService.java [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+    MirrorNodeService(MirrorNodeGateway mirrorNodeGateway) {
+        this.mirrorNodeGateway = mirrorNodeGateway;
+    }
+
+    /**
+     * Get AccountId num from mirror node using evm address.
+     *
+     * @param evmAddress
+     * @return
```
Comment (agadzhalov) [inline sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeService.java:44]: minor suggestion: the last line of these doc comments can be updated to say what the method actually returns
```suggestion
     * @return account id
```

**Review by agadzhalov — APPROVED** [review]
LGTM

File: sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeService.java [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        JsonArray tokens = accountTokensResponse.get("tokens").getAsJsonArray();
+
+        List<TokenRelationship> tokenRelationships = tokens.asList().stream().map(jsonElement -> {
+            var jsonObject = jsonElement.getAsJsonObject();
+            var tokenId = jsonObject.get("token_id").getAsString();
+            var balance = jsonObject.get("balance").getAsLong();
+            var kycStatus = jsonObject.get("kyc_status").getAsString();
+            var freezeStatus = jsonObject.get("freeze_status").getAsString();
+            var decimals = jsonObject.get("decimals").getAsInt();
+            var automaticAssociation = jsonObject.get("automatic_association").getAsBoolean();
```
[the remark this answers is not in the document]
Comment (thenswan) [reply sdk/src/main/java/com/hedera/hashgraph/sdk/MirrorNodeService.java:167]: Thanks @0xivanov, I referenced in a follow up issue #1815

**Pull request merged**